src/editor/controls/GitHubRepositoryControl.tsx

```diff
@@ -0,0 +1,119 @@
+import {BaseControl, Button, TextControl,} from '@wordpress/components';
+import {useEffect, useRef} from '@wordpress/element';
+import {__} from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+
+interface GitHubRepositoryControlProps {
+    value: string;
+    onChange: (value: string) => void;
+}
+
+export const GitHubRepositoryControl = ({value, onChange, onCodeFetched}: GitHubRepositoryControlProps) => {
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+        if (isValidUrl(value)) {
+            fetchFile(value).then((code) => {
+                if (value.indexOf("#L")) {
+                    onCodeFetched({ code, lineNumbers: extractLineNumbers(value) });
+                } else {
+                    onCodeFetched({ code });
+                }
+            });
+
+            inputRef.current
+                ?.querySelector('input')
+                ?.classList.remove('cbp-input-error');
+        } else {
+            inputRef.current
+                ?.querySelector('input')
+                ?.classList.add('cbp-input-error');
+        }
+    }, [value]);
+    return (
+        <BaseControl id="code-block-pro-show-highlighting">
+            <div ref={inputRef}>
+                <TextControl
+                    spellCheck={false}
+                    autoComplete="off"
+                    type="text"
+                    data-cy="github-repository-link"
+                    label={__('Github Repository Link', 'code-block-pro')}
+                    help={__(
+                        'The link to your file. Supports github.com links, gists, and raw content links.',
```

Comment:
"raw content links" might let the user think they can paste in any code url (which might be ok actually - but then we should check the mime type that I mentioned not checking)